STYLE_GUIDE.md

```diff
@@ -0,0 +1,183 @@
+
+## Naming Conventions
+Give your functions, components and files **speaking/self-explanatory names** (e.g. `getCollectionById()`, `KeyboardShortcuts.vue`, `setIdentity.ts`).
+With a few exceptions, code and comments should be written in **english** only.
+
+### Nuxt/Vue-specific Files
+- **Pages** and **layouts** use **kebab-case** (`series-insight.vue`)
+- **All other components** (in folder `/components`) use **PascalCase** (`CookieBanner.vue`)
+
+### Others
+- **Folders** are written in **lowercase** (`components/spotlight`)
+- **Typescript**, **Javascript** and **GraphQL** files use **camelCase** (`globalVariables.ts`, `getKey.js`, `collectionById.graphql`)
+- **SCSS** files use **kebab-case** (`initial-variables.scss`)
+- **JSON** files use **snake_case** (`all_lang.json`) while **Markdown** files use **SCREAMING_SNAKE_CASE** (`CONTRIBUTING.md`)
+
+## SFC Conventions
+### Skeleton
+99% of the time your SCSS should be **scoped**, so it won't bleed outside of your component and pollute the global namespace!
+```vue
+<template>
+    <div>
+        ...
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'nuxt-property-decorator'
+import { ComponentOne } from '@/components/ComponentOne.vue'
+
+interface InterfaceName {
+  property: type
+}
+
+@Component({
+  components: {
+    ComponentOne
+  },
+})
+export default class ComponentName extends Vue {}
+</script>
+
+<style scoped lang="scss"></style>
+```
+
+### Property Decorators
+We rely on the package 'nuxt-property-decorator', hence, we urge you to comply with the [Nuxt Class Component Syntax](https://github.com/nuxt-community/nuxt-property-decorator/)
+```typescript
+import {
+  Component,
+  Inject,
+  Model,
+  Prop,
+  Provide,
+  Vue,
+  Watch,
+} from "nuxt-property-decorator"
+
+@Component({})
+export class MyComponent extends Vue {
+  @Inject() foo!: string
+
+  @Model("change") checked!: boolean
+
+  @Prop() propA!: number
+
+  @Provide() foo = "foo"
+
+  @Watch("child")
+  onChildChanged(val: string, oldVal: string) {}
+}
+```
+
+
+### Using Components In Templates
+Custom components and prop bindings should be used like this
+```vue
+<MyCustomComponent :property-name="doSomething">
+    {{ someProperty }}
+</MyCustomComponent>
+```
+
+Use shorthands for vue attributes
+- `:href` instead of `v-bind:href`
+- `@click` instead of `v-on:click`
+- ...
+
+
+### Fetching Data
+Though we haven't yet transitioned most of our data fetching logic to Nuxt lifecycles, the following syntax should be considered best practice:
+```typescript
+// pages
+import { Component } from 'nuxt-property-decorator'
+
+@Component({
+  async asyncData({ app, $config, params }) {
+    const res = await app?.apolloProvider?.clients[$config.prefix].query({
+      query: queryGql,
+      variables: {
+        id: params.id
+      },
+    })
+
+    return {
+      data: res.data,
+      total: res.total,
+    }
+  }
+})
```

Comment:
asyncData could be defined as function insside the component

Reply:
I think its actually very convenient to be able to use nuxt hooks outside the component body